examples/1-hello-world/README.md

```diff
@@ -0,0 +1,134 @@
+# Hello World
+
+This is very simple `bin` crate generated by `cargo new`, with no additional
+modification. It is built using the latest stable Rust, as provided by
+[nixpkgs-mozilla](https://github.com/mozilla/nixpkgs-mozilla).
+
+## How it was created
+
+1. Install the `cargo2nix` tool according to the [official instructions], if you
+   haven't already.
+2. Create a new `bin` project with `cargo new hello-world`. If the Rust
+   toolchain is not installed on your system, you could use `nix-shell` like so:
+   ```bash
+   nix-shell -p cargo --run 'cargo new hello-world'
+   ```
+3. Run `cargo generate-lockfile` to create a `Cargo.lock`, if one is not
+   already present.
+4. Run `cargo2nix -f` at the project root to generate a `Cargo.nix`.
+5. Write the `default.nix`, and optionally a `shell.nix`.
+
+[official instructions]: ../../README.md#install
+
+## Building
+
+To compile the `hello-world` binary with Nix, run:
+
+```bash
+nix-build -A package
+```
+
+This will create a `result` symlink in the current directory with the following
+structure:
+
+```text
+/nix/store/cjwyldan7dfzp6494pl5vsf0vlaa7dx2-crate-hello-world-0.1.0
+├── .cargo-info
+├── bin
+│   └── hello-world
+└── lib
+    └── .link-flags
+```
+
+Running the `hello-world` binary will print the following message to the screen:
+
+```text
+$ ./result/bin/hello-world
+Hello, world!
+```
+
+## Running tests
+
+To build and run the test binaries for this project, run:
+
+```bash
+nix-build -A ci
```

Comment:
Is the user being asked to create this project and then compare their result or just to consume our result?  Going through the motions themselves is likely quite valuable.  At the end say, _The example output of this flow is over in `/sample/`_

It's hard to understand a c2n project without going through the bootstrap IMO.   `ci` just exists out of nowhere at this point.

Okay, I see.  Earlier the instruction says:

> write the default.nix

https://knowyourmeme.com/memes/how-to-draw-an-owl  :grinning: 

Adding a stripped down default.nix (for some reason commentary makes the c2n one scary at first glance) will both fit into the instructions succinctly and lead the user to understand where these build attributes are showing up from